zio-flow/shared/src/test/scala/zio/flow/mock/MockedOperation.scala

```diff
@@ -43,19 +43,18 @@ object MockedOperation {
   ) extends MockedOperation {
     override def matchOperation[R1, A1](operation: Operation[R1, A1], input: R1): (Option[Match[A1]], MockedOperation) =
       operation match {
-        case Operation.Http(url, method, headers, _, _) =>
+        case Operation.Http(url, api) =>
           // TODO: check R1 and A1 types too
+          // TODO: check headers as well
           val m =
-            urlMatcher.run(url) && methodMatcher.run(method) && headersMatcher.run(headers) && inputMatcher.run(
+            urlMatcher.run(new URI(url)) && methodMatcher.run(api.method.toString()) && inputMatcher.run(
```

Comment:
Let's modify the matcher's types instead of converting to `URI` and `toString` here. I think if we make it closer to the thing we mock, it's easier to use